### PR TITLE
Delete Demo interfaces

### PR DESCRIPTION
**Pull request by Arrgentum**
## Whats added:
- Removing unnecessary interfaces:
-- DemoAdditionalParams.kt
-- DemoRequest.kt
-- DemoResult.k

## First path issue #1762

### CODE REVIEW

**Comment by nulls** [comment]
@sanyavertolet why do we remove these interfaces?

**Comment by sanyavertolet** [comment]
> @sanyavertolet why do we remove these interfaces?

As we want to support a configurable demo, we are not going to implement those interfaces. We are going to use one unified data class instead of each interface.

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
  */
 @Serializable
-data class DiktatAdditionalParams(
-    val mode: DiktatDemoMode = DiktatDemoMode.WARN,
-    val tool: DiktatDemoTool = DiktatDemoTool.DIKTAT,
-    val config: List<String> = defaultDiktatConfig,
+data class DemoRunRequest(
+    val codeLines: List<String>,
+    val mode: DemoMode?,
+    val config: List<String>?,
```
Comment (sanyavertolet) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt]: Maybe make it `null` by default?

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt [hunk 2]
```diff
@@ -118,5 +121,6 @@ data class DiktatAdditionalParams(
             |- name: COMPLEX_EXPRESSION
             |  enabled: true
         """.trimMargin().split("\n")
+        val diktat = DemoRunRequest(emptyList(), DemoMode.WARN, defaultDiktatConfig)
```
Comment (sanyavertolet) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt:124]:
```suggestion
        val diktatDemoRunRequest = DemoRunRequest(emptyList(), DemoMode.WARN, defaultDiktatConfig)
```

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DiktatDemoController.kt [hunk 3]
```diff
@@ -23,13 +25,13 @@ class DiktatDemoController(
...
+     * @return [DemoResult]
      */
     @PostMapping("/run")
     fun runCheckDemo(
-        @RequestBody diktatRunRequest: DiktatRunRequest,
-    ): Mono<DiktatDemoResult> = blockingToMono {
-        diktatDemoService.launch(diktatRunRequest.codeLines, diktatRunRequest.params)
+        @RequestBody demoRunRequest: DemoRunRequest,
+    ): Mono<DemoResult> = blockingToMono {
+        diktatDemoService.launch(demoRunRequest.codeLines, demoRunRequest)
```
Comment (sanyavertolet) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DiktatDemoController.kt:35]: Is there any need to pass `codeLines` twice?

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/service/AbstractDemoService.kt [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
+     * Run demo on [demoFileLines] with [runRequest] and return result as [DemoResult]
      *
      * @param demoFileLines list of lines of input file that will be used for demo
-     * @param diktatAdditionalParams additional params as [DiktatAdditionalParams]
-     * @return report as [DiktatDemoResult]
+     * @param runRequest additional params as [DemoRunRequest]
+     * @return report as [DemoResult]
      */
-    abstract fun launch(demoFileLines: List<String>, diktatAdditionalParams: P? = null): R
+    abstract fun launch(demoFileLines: List<String>, runRequest: DemoRunRequest = DemoRunRequest.empty): DemoResult
```
Comment (sanyavertolet) [inline save-demo/src/main/kotlin/com/saveourtool/save/demo/service/AbstractDemoService.kt:20]: do we need to pass `demoFileLines` twice?

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt [hunk 5]
```diff
@@ -215,9 +202,7 @@ private fun diktatDemoComponent() = FC<DiktatDemoComponentProps> { props ->
                         setDiktatRunRequest { runRequest ->
                             (event.target.asDynamic()["result"] as String?)?.let {
                                 runRequest.copy(
-                                    params = runRequest.params.copy(
-                                        config = (it.split("\n"))
-                                    )
+                                    config = (it.split("\n"))
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt:205]: do we need those parentheses?

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt (hunk 1 above)
Comment (Arrgentum) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt:15]: Done

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt (hunk 2 above)
Comment (Arrgentum) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt]: Done

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DiktatDemoController.kt (hunk 3 above)
Comment (Arrgentum) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/controller/DiktatDemoController.kt]: Done

File: save-demo/src/main/kotlin/com/saveourtool/save/demo/service/AbstractDemoService.kt (hunk 4 above)
Comment (Arrgentum) [reply save-demo/src/main/kotlin/com/saveourtool/save/demo/service/AbstractDemoService.kt]: Done

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt (hunk 5 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt]: Done

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt [hunk 6]
```diff
@@ -221,7 +221,7 @@ private fun diktatDemoComponent() = FC<DiktatDemoComponentProps> { props ->
                     }
                 }
                 fontAwesomeIcon(icon = faUpload)
-                val uploadOrReplace = if (diktatRunRequest.config
+                val uploadOrReplace = if (diktatRunRequest.config.orEmpty()
```
Comment (sanyavertolet) [inline save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt:224]: Maybe `isNullOrEmpty`?

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt:15]: no, you didn't get it
```suggestion
    val mode: DemoMode? = null,
    val config: List<String>? = null,
```

File: save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt (hunk 6 above)
Comment (Arrgentum) [reply save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/DiktatDemoComponent.kt]: Done

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt (hunk 1 above)
Comment (Arrgentum) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/demo/DemoRunRequest.kt]: Done

**Review by sanyavertolet — APPROVED** [review]
lgtm

**Pull request merged**